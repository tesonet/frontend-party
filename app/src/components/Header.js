import React, { PropTypes } from 'react';
import FlatButton from './FlatButton';
import './Header.scss';

const Header = ({ onButtonClick }) => (
  <div className="header">
    <img src={require('../../public/images/logo.png')} alt="testio" />
    <FlatButton
      icon="log-out"
      onClick={onButtonClick}
    >
       Logout
     </FlatButton>
  </div>
);

Header.propTypes = {
  onButtonClick: PropTypes.func.isRequired,
};

export default Header;
