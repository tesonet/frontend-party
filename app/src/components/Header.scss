.header {
  flex: 0 0 120px;
  width: 100%;
  display: flex;
  justify-content: space-between;
  align-items: flex-start;
  padding: 40px 10px;
}
