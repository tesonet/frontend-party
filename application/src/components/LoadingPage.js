import React, { Component } from 'react';
import './styles/LoadingPageStyle.css';

class LoadingPage extends Component {
  render() {
    return(
      <div className='page'>
        <div className='container'>
        <div className="spinner">
          <div className="bounce1"></div>
          <div className="bounce2"></div>
          <div className="bounce3"></div>
        </div>
        </div>
      </div>
    )
  }
}

export default LoadingPage;
