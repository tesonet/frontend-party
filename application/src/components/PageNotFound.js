import React, { Component } from 'react';
import { connect } from 'react-redux';
import { Redirect } from 'react-router-dom';
import './styles/PageNotFoundStyle.css';

class PageNotFound extends Component {

  constructor() {
    super()
    this.state = { redirect: false }
  }

  componentDidMount(){
    setTimeout(() => {
      this.setState({ redirect: true })
    }, 5000)
}

  render() {
    if (this.state.redirect) {
      return <Redirect to='/'/>
    }
    return (
      <div className='page'>
        <div className='container'>
          <h1>404 - Page Not Found</h1>
          <p>Sorry, the page you were looking for does not exist.</p>
          <p>You will be redirected to {(this.props.login.token && this.props.login.userAuthenticated) ? "servers" : "login"} page in a few seconds</p>
        </div>
      </div>
    )
  }
}

const mapStateToProps = (state) => {
  return {
    login: state.LoginReducer
  }
}



export default connect(mapStateToProps)(PageNotFound);
