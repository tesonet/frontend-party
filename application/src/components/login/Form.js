import React, { Component } from 'react';
import { connect } from 'react-redux';
import { Login } from '../../actions/index';
// import { FormStyle } from './styles/FormStyle';

import { Field, reduxForm, SubmissionError } from 'redux-form'

class Form extends Component {

    submit = (values) => {

      let error = {},
          isError = false;

      if(values.username == undefined) {
        error.username = 'Please enter your username';
        isError = true;
      }

      if(values.password == undefined) {
        error.password = 'Please enter your password';
        isError = true;
      }

      if (isError) {
        throw new SubmissionError(error)
      } else {
        this.props.Login(values);
        console.log(this.props.login);
        //then dispatch servers?
      }
    }

    renderField = ({type, className, label, input, placeholder, meta: {touched, error} }) => (

        <div className='form-group'>
          <input {...input} className={className} type={type} placeholder={placeholder}/>
          {touched && error &&
           <span className="error">{error}</span>}
        </div>
    );

    render() {
        return (
          <div className="container">
            <form onSubmit={this.props.handleSubmit(this.submit)}>
              <Field name="username" className='form-control'
              component={this.renderField} type="text"
              placeholder='Username'/>
              <Field name="password" className='form-control'
              component={this.renderField} type="password"
              placeholder='Password'/>
              <button type="submit" className='btn'>Login</button>
            </form>
            <span>{this.props.login.errorMessage ? this.props.login.errorMessage : ''}</span>
          </div>
        )
      }
    }

  Form = reduxForm({
    form: 'login',
    destroyOnUnmount: false
  })(Form)

  const mapStateToProps = (state) => {
    return {
      login: state.LoginReducer
    }
  }

  const mapDispatchToProps = (dispatch) => {
  return {
    Login: (values) => {
      dispatch(Login(values));
    }
  }
}

export default connect(mapStateToProps, mapDispatchToProps)(Form);
