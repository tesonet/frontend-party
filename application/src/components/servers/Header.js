import React, { Component } from 'react';
import { connect } from 'react-redux';
import { logout } from '../../actions/index';
import logoDark from '../../assets/logo-dark.png';
import './styles/HeaderStyle.css';

class LogoutIcon extends Component {
  render() {
    return(
      <svg className="logout-icon" width="16" height="16" viewBox="0 0 16 16">
        <path id="ico-logout" d="M3027,54l-4,4,4,4V59h6V57h-6V54Zm11,12h-9a0.945,0.945,0,0,1-1-1V62h2v2h7V52h-7v2h-2V51a0.945,0.945,0,0,1,1-1h9a0.945,0.945,0,0,1,1,1V65A0.945,0.945,0,0,1,3038,66Z" transform="translate(-3023 -50)"/>
      </svg>
    )
  }
}

class Header extends Component {
  render() {
    return(
      <div className='header'>
          <div className='top row'>
            <div className='col-xs-6'><img src={logoDark} alt={'Testio logo'} /></div>
            <div className='col-xs-6 right log-out'>
              <div className='' onClick={() => {this.props.logout()}}>
            <LogoutIcon />
            <span>Logout</span></div></div>
          </div>
      </div>
    )
  }
}

const mapStateToProps = (state) => {
  return {
    login: state.LoginReducer,
    servers: state.ServersReducer
  }
}

const mapDispatchToProps = (dispatch) => {
return {
  logout: () => {
    dispatch(logout());
  }
}
}

export default connect(mapStateToProps, mapDispatchToProps)(Header);
