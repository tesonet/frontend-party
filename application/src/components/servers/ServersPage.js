import React, { Component } from 'react';
import { connect } from 'react-redux';
import { fetchServers } from '../../actions/index';
import { uniqueId, sortBy } from 'lodash';
import Header from './Header';
import { Scrollbars } from 'react-custom-scrollbars';
import './styles/ServersPageStyle.css'

class ServersPage extends Component {
  constructor(props) {
    super(props);
    this.state = this.props;
  }

  componentWillMount() {
  if (this.props.login.userAuthenticated && this.props.login.token) {
      this.props.fetchServers();
    }
    console.log(JSON.parse(localStorage.getItem('reduxPersist:LoginReducer')).token);
  }

  listServers() {
    return this.props.servers.servers.map((server) => {
      server.id = uniqueId('ID-');
      return (
          <div className='list row' key={server.id}>
                <div className='col-xs-6'>{server.name}</div>
                <div className='right col-xs-6'>{server.distance} km</div>
          </div>
      )
    })
  }

  sortServers(field) {
    let serverList = this.props.servers.servers;
    this.props.servers.servers = sortBy(serverList, field);
    this.setState(this.props);
  }

  render() {
    return(
      <div>
          <Header />
          <div className='info row'>
              <div className='col-xs-6' onClick={() => {this.sortServers('name')}}>Server</div>
              <div className='right col-xs-6' onClick={() => {this.sortServers('distance')}}>Distance</div>
          </div>
            <div className='servers container-fluid'>
              <Scrollbars autoHide={true}>
                {this.listServers()}
              </Scrollbars>
            </div>
      </div>
    )
  }
}

const mapStateToProps = (state) => {
  return {
    login: state.LoginReducer,
    servers: state.ServersReducer
  }
}

const mapDispatchToProps = (dispatch) => {
  return {
    fetchServers: () => {
      dispatch(fetchServers());
    }
  }
}

export default connect(mapStateToProps, mapDispatchToProps)(ServersPage);
