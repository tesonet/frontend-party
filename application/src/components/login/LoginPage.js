import React, { Component } from 'react';
import Form from './Form';

class LoginPage extends Component {

  render() {
    return(
      <div className='container'>
        <h1>TESTIO.</h1>
        <Form />
      </div>
    );
  }
}

export default LoginPage;
