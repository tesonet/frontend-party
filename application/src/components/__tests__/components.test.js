import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import LoginPage from '../login/LoginPage';
import Form from '../login/Form';
import PageNotFound from '../PageNotFound';
import LoadingPage from '../LoadingPage';
import Header from '../servers/Header';
import ServersPage from '../servers/ServersPage';
import configureStore from 'redux-mock-store';
import store from '../../Store';

test('Login page should render as expected',() => {
  const component = shallow(<LoginPage />);
  let tree = toJson(component);
  expect(tree).toMatchSnapshot();
});

test('Form should render as expected',() => {
  const component = shallow(<Form store={store} onSubmit={() => console.log('onSubmit works')}/>);
  let tree = toJson(component);
  expect(tree).toMatchSnapshot();

  tree.props.onSubmit();

  tree = toJson(component);
  expect(tree).toMatchSnapshot();
});

test('404 page should render as expected',() => {
  const component = shallow(<PageNotFound store={store}/>);
  let tree = toJson(component);
  expect(tree).toMatchSnapshot();
});

test('Loading page should render as expected',() => {
  const component = shallow(<LoadingPage />);
  let tree = toJson(component);
  expect(tree).toMatchSnapshot();
});

test('Header should render as expected',() => {
  const component = shallow(<Header store={store}/>);
  let tree = toJson(component);
  expect(tree).toMatchSnapshot();
})

test('Server list should render as expected', () => {
  const component = shallow(<ServersPage store={store}/>);
  let tree = toJson(component);
  expect(tree).toMatchSnapshot();
})
