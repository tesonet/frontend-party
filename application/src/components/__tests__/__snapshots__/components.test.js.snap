// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`404 page should render as expected 1`] = `
<PageNotFound
  dispatch={[Function]}
  login={
    Object {
      "errorMessage": false,
      "pending": false,
      "token": false,
      "userAuthenticated": false,
    }
  }
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
  storeSubscription={
    Subscription {
      "listeners": Object {
        "notify": [Function],
      },
      "onStateChange": [Function],
      "parentSub": undefined,
      "store": Object {
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
        Symbol(observable): [Function],
      },
      "unsubscribe": null,
    }
  }
/>
`;

exports[`Form should render as expected 1`] = `
<ReduxForm
  Login={[Function]}
  login={
    Object {
      "errorMessage": false,
      "pending": false,
      "token": false,
      "userAuthenticated": false,
    }
  }
  onSubmit={[Function]}
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
  storeSubscription={
    Subscription {
      "listeners": Object {
        "notify": [Function],
      },
      "onStateChange": [Function],
      "parentSub": undefined,
      "store": Object {
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
        Symbol(observable): [Function],
      },
      "unsubscribe": null,
    }
  }
/>
`;

exports[`Form should render as expected 2`] = `
<ReduxForm
  Login={[Function]}
  login={
    Object {
      "errorMessage": false,
      "pending": false,
      "token": false,
      "userAuthenticated": false,
    }
  }
  onSubmit={[Function]}
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
  storeSubscription={
    Subscription {
      "listeners": Object {
        "notify": [Function],
      },
      "onStateChange": [Function],
      "parentSub": undefined,
      "store": Object {
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
        Symbol(observable): [Function],
      },
      "unsubscribe": null,
    }
  }
/>
`;

exports[`Header should render as expected 1`] = `
<Header
  login={
    Object {
      "errorMessage": false,
      "pending": false,
      "token": false,
      "userAuthenticated": false,
    }
  }
  logout={[Function]}
  servers={
    Object {
      "errorMessage": false,
      "pending": false,
      "servers": Array [],
    }
  }
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
  storeSubscription={
    Subscription {
      "listeners": Object {
        "notify": [Function],
      },
      "onStateChange": [Function],
      "parentSub": undefined,
      "store": Object {
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
        Symbol(observable): [Function],
      },
      "unsubscribe": null,
    }
  }
/>
`;

exports[`Loading page should render as expected 1`] = `
<div
  className="page"
>
  <div
    className="container"
  >
    <div
      className="spinner"
    >
      <div
        className="bounce1"
      />
      <div
        className="bounce2"
      />
      <div
        className="bounce3"
      />
    </div>
  </div>
</div>
`;

exports[`Login page should render as expected 1`] = `
<div
  className="loginPage"
>
  <div
    className="container"
  >
    <img
      alt="Testio logo"
      className="logo"
      src="logo-white.png"
    />
    <Connect(ReduxForm) />
  </div>
</div>
`;

exports[`Server list should render as expected 1`] = `
<ServersPage
  fetchServers={[Function]}
  login={
    Object {
      "errorMessage": false,
      "pending": false,
      "token": false,
      "userAuthenticated": false,
    }
  }
  servers={
    Object {
      "errorMessage": false,
      "pending": false,
      "servers": Array [],
    }
  }
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
  storeSubscription={
    Subscription {
      "listeners": Object {
        "notify": [Function],
      },
      "onStateChange": [Function],
      "parentSub": undefined,
      "store": Object {
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
        Symbol(observable): [Function],
      },
      "unsubscribe": null,
    }
  }
/>
`;
